backend/src/authorization/guards/roles.guard.ts

```diff
@@ -0,0 +1,24 @@
+import { Injectable, CanActivate, ExecutionContext } from '@nestjs/common';
+import { Reflector } from '@nestjs/core';
+import { Role } from '@prisma/client';
+import { RequestWithUser } from 'src/authentication/types/request-with-user.interface';
+import { ROLES_KEY } from '../decorators/roles.decorator';
+
+@Injectable()
+export class RolesGuard implements CanActivate {
+  constructor(private readonly reflector: Reflector) {}
+
+  canActivate(context: ExecutionContext): boolean {
```

Comment:
here is the actual logic for restricting endpoints based on what information was provided by the `@Roles` decorator